ethcore/engines/authority-round/src/lib.rs

```diff
@@ -1116,6 +1110,18 @@ impl AuthorityRound {
 		let tx_request = TransactionRequest::call(contract_addr, data).gas_price(U256::zero()).nonce(nonce);
 		Ok(vec![full_client.create_transaction(tx_request)?])
 	}
+
+	/// Returns the reference to the client, if registered.
+	fn client_arc<'a, T>(&self, opt_error_msg: T) -> Result<Arc<dyn EngineClient>, EngineError>
```

Comment:
Maybe a better name?

```suggestion
	fn upgrade_client_or<'a, T>(&self, opt_error_msg: T) -> Result<Arc<dyn EngineClient>, EngineError>
```

Reply:
Done.